lib/client-options.js

```diff
@@ -87,7 +90,13 @@ function extend(baseOptions, userOptions) {
     throw new TypeError('queryOptions not defined in options');
   }
   if (!options.protocolOptions) {
-    throw new TypeError('protocolOptions');
+    throw new TypeError('protocolOptions not defined in options');
+  }
+  if (!options.socketOptions) {
```

Comment:
If not specified, it sets the defaults, this is just a validation for our default settings.